api/nodecontrol/client.go

```diff
@@ -118,6 +134,40 @@ func (c *ControlAPIClient) FindWorkload(_type, namespace, workloadId string) (*n
 	return resp, nil
 }
 
+func (c *ControlAPIClient) AuctionDeployWorkload(namespace string, nodeTags map[string]string, req nodegen.StartWorkloadRequestJson) (*nodegen.StartWorkloadResponseJson, error) {
+	auctionResults, err := c.Auction(nodeTags)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(auctionResults) == 0 {
+		c.logger.Info("no nodes available for deployment")
+		return nil, nil
+	}
+
+	nodeIdx := rand.IntN(len(auctionResults))
+	// This "NodeID" is acutally the BidderID
+	node := auctionResults[nodeIdx].NodeId
+
+	req_b, err := json.Marshal(req)
+	if err != nil {
+		return nil, err
+	}
+
+	msg, err := c.nc.Request(models.AuctionDeployRequestSubject(namespace, node), req_b, DefaultRequestTimeout)
```

Comment:
this shouldn't be `node` here. This is the auction ID that we chose, _not_ the node ID.